### PR TITLE
Add CatalogClient functional tests

### PR DESCRIPTION
**Pull request by LiubovDidkivska**
Add versions list CatalogClient tests.
Change functional CatalogClient tests
to use mock server.
Add helper functionality for mock server.
    
Relates-To: OLPEDGE-1606
    
Signed-off-by: Liubov Didkivska <ext-liubov.didkivska@here.com>

### CODE REVIEW

File: tests/utils/mock-server-client/Client.h [hunk 1]
```diff
@@ -42,15 +41,15 @@ const auto kClearPath = "/mockserver/clear";
...
 
 class Client {
  public:
   explicit Client(olp::client::OlpClientSettings settings);
 
   void MockResponse(const std::string& method_matcher,
                     const std::string& path_matcher,
-                    const std::string& response_body, bool unlimited = false);
+                    const std::string& response_body,
+                    uint16_t status_code = 200, bool unlimited = false);
```
Comment (mykhailo-kuchma) [inline tests/utils/mock-server-client/Client.h:52]: We should use http::HttpStatusCode

File: tests/functional/utils/MockServerHelper.cpp [hunk 2]
```diff
@@ -84,24 +76,70 @@ void MockServerHelper::MockAuth() {
...
+
+void MockServerHelper::MockGetVersionInfosResponse(
+    olp::dataservice::read::model::VersionInfos data) {
+  MockGetResponse(std::move(data),
+                  "/metadata/v1/catalogs/" + catalog_ + "/versions");
+}
+
+void MockServerHelper::MockGetCatalogResponse(
+    olp::dataservice::read::model::Catalog data) {
+  MockGetResponse(std::move(data), "/config/v1/catalogs/" + catalog_);
```
Comment (andescu) [inline tests/functional/utils/MockServerHelper.cpp:111]: Same here.

File: tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp [hunk 3]
```diff
@@ -118,4 +118,30 @@ TEST_F(VersionedLayerClientTest, GetPartitions) {
...
+          hrn, "testlayer", boost::none, *settings_);
+
+  auto request = olp::dataservice::read::PartitionsRequest();
+  auto future = catalog_client->GetPartitions(request);
+  auto partitions_response = future.GetFuture().get();
+  ASSERT_EQ(olp::client::ErrorCode::BadRequest,
+            partitions_response.GetError().GetErrorCode());
+  ASSERT_EQ("Bad request", partitions_response.GetError().GetMessage());
+  EXPECT_TRUE(mock_server_client_->Verify());
+}
```
Comment (andescu) [inline tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp:145]: This doesn't seem related to the ListVersions functionality. Why did you add it in the scope of this ticket?

File: tests/functional/olp-cpp-sdk-dataservice-read/CatalogClientTest.cpp [hunk 4]
```diff
@@ -142,6 +212,74 @@ TEST_P(CatalogClientTest, GetCatalog) {
...
 
+TEST_F(CatalogClientTest, GetVersionsList) {
+  const auto catalog = olp::client::HRN::FromString(kTestHrn);
+  auto client = std::make_unique<olp::dataservice::read::CatalogClient>(
+      catalog, settings_);
+  {
+    SCOPED_TRACE("Get versions list online");
+    {
+      mock_server_client_->MockAuth();
+      mock_server_client_->MockTimestamp();
```
Comment (andescu) [inline tests/functional/olp-cpp-sdk-dataservice-read/CatalogClientTest.cpp:223]: I don't think these are necessary anymore. Timestamp should be disabled by default.

File: tests/functional/olp-cpp-sdk-dataservice-read/CatalogClientTest.cpp [hunk 5]
```diff
@@ -69,46 +89,87 @@ std::ostream& operator<<(std::ostream& os, const CacheType cache_type) {
...
+  olp::dataservice::read::model::VersionInfos infos;
+  auto size = end - start;
+  if (size <= 0) {
+    return infos;
+  }
+  std::vector<olp::dataservice::read::model::VersionInfo> versions_vect(size);
+  for (size_t i = 0; i < versions_vect.size(); i++) {
+    versions_vect[i].SetVersion(++start);
+    versions_vect[i].SetTimestamp(1000 * start);
+    std::vector<olp::dataservice::read::model::VersionDependency> dependencyes(
```
Comment (andescu) [inline tests/functional/olp-cpp-sdk-dataservice-read/CatalogClientTest.cpp:103]:
```suggestion
    std::vector<olp::dataservice::read::model::VersionDependency> dependencies(
```
